lib/middleware.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+// Chain is essentially a linked list of middlewares to run in order.
+function Chain() {
+  var self = this;
+  var middlewares = [];
+
+  this.getMiddlewares = function() {
+    return middlewares;
```

Comment:
Should we `.slice()` this to prevent any mutations?

Reply:
Very good idea!